concepts/tail-call-recursion/about.md

```diff
@@ -0,0 +1,64 @@
+# About
```

Comment:
This is a great explanation

Reply:
I'm sitting on the shoulders of giants (@neenjaw)